docs/DeveloperGuide.md

```diff
@@ -1,38 +1,536 @@
+and new values (description or answer).
 
-## Design & implementation
+- **Validation**: Implement validation logic to ensure that the user input is correctly formatted and contains valid 
+information.
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+- **Edit Operation**: Implement the logic to execute the edit operation based on the provided criteria. This typically 
+involves invoking methods in the QuestionList and Question classes to update the question's description or answer.
 
+- **Data Persistence**: If necessary, update the data storage to save the changes. In the provided code, the 
+`dataStorage.updateData(questions)` method is used to save changes to the question list.
 
-## Product scope
-### Target user profile
+- **Error Handling**: Handle any exceptions or errors that may occur during the edit operation and provide appropriate 
+feedback to the user.
 
-{Describe the target user profile}
+### Start Command - Start Quiz
 
-### Value proposition
+#### Brief Description of Start Command
 
-{Describe the value proposition: what problem does it solve?}
+The start quiz feature allows users to start quizzing themselves with customizable characters to define which modules
+to quiz themselves on alongside whether to randomize the questions or use their pre-defined question order.
+- `start /[quiz mode] [start details] /[qn mode]`
+
+#### Class Structure of Start Command
+
+![commandStartSequence.png](UML%2FCommands%2FcommandStartSequence.png)
+
+#### Implementation of Start Command
+
+The start quiz mechanism is facilitated by CommandStart under package quizhub.command. The class utilises methods from `quizhub.questionlist.QuestionList`.  It extends Command with 2 new prompts (`/[quiz mode]` and `/[qn mode]`) and 1 user input field (`/[start details]`). It implements the following operations:
+
+`/[quiz mode]`
+1. 2 configurations - `/module` and `/all`
+2.	`/module` must be followed by a category name to retrieve questions from that specific category
+- The method categoriseListByModules from the package `quizhub.questionlist.QuestionList` will be called to retrieve the questions by that are listed within the specified category from the storage list
+- `/all` will not require any input from `/[start details]`
+3.	The method `getAllQns()` from package `quizhub.questionlist.QuestionList` will be called to retrieve all questions from the storage list.
+
+`/[qn mode]`
+1.	2 configurations - `/random` and `/normal`
+2.	This prompt is activated after defining `/[quiz mode]` and `/[start details]`
+3.	/random will randomize the list of questions using `java.util.Collections.shuffle` and store it within a temporary array to prevent tempering with the original array in Storage
+4.	`/normal` will not requirwritee any further actions, using the previously generated list as specified by `/[quiz mode]` and `/[start details]`
```

Comment:
Perhaps /random can be written in a code block? I believe there is a typo "requirwritee".